### DISCUSSION TITLE
Is it possible to render a front-end form based on a content type via a driver?

### DISCUSSION THREAD

**Discussion by natesoch** [OrchardCMS/OrchardCore#13978] [General]
Basically, I currently have a custom MVC form with a hardcoded POST action. What I am doing is a bit specific. I have two content types, one representing a client (which shows what checkboxes to render on the form for a logged in user) and a content type for the form submission itself, which has the list of all the available checkboxes. What I am doing is I am only rendering the checkboxes on the form the logged in user has access to (based on their client type). While this works fine, if I want to add a new checkbox I need to add it to the  content types and add a few lines to my source code (I have a dictionary that maps the selected items to the properties on the content part so I can mark them as true).

What I want to be able to do is render the front-end form similar to the content item editor seen in the Orchard admin. I'd need to be able to hide some fields (e.g.  only show the allowed permissions boolean field, etc.). If a new checkbox is added to the content type, then it should render it based on that without having to add lines of code like I do now. Is this something that is supported by Orchard Core? Or even more, is this feasible to do? I would appreciate any insight or links to documentation that might help with this—I haven't been able to find anything.

**Comment by DrewBrasher** [comment]
I was also curious if the admin editor could be used on the front end in this discussion: https://github.com/OrchardCMS/OrchardCore/discussions/13915

I'm still not really sure if it's possible.

**Comment by ns8482e** [comment]
Yes It's possible - you can create implement your own IThemeSelecor that returns the theme based on ContentType being served.
One example is Login page - where the theme is selected based on Settings - check code for User module
